roles/openshift_metrics/handlers/main.yml

```diff
@@ -1,18 +1,12 @@
 ---
-- name: restart master api
-  systemd: name={{ openshift_service_type }}-master-api state=restarted
+- name: restart master
+  command: /usr/bin/master-restart "{{ item }}"
+  with_items:
+  - api
```

Comment:
Unfortunately, api is not 100% up when it reports back to systemd that it has started.  This can cause the controller to fail; that's why we retry the controllers.

Reply:
The controllers are now restarted by the kubelet, they will eventually converge.  The only thing we should be doing is looping waiting for things to happen later.

Reply:
Put another way, there is no scenario where the controllers shouldn't eventually start

Reply:
On regular systemd (which we plan to remove in 3.10), I could have the master-restart script ignore the restart failure as long as systemd is guaranteed to continue starting it.  If we issue a direct restart, will systemd continue to attempt to start it if the first start fails?

Reply:
I'm unsure if the controller was tyring to restart itself.  I think the issue we were having was that ansible was unable to successful start he controller on the first attempt, thus we just put a retry on the restart job.

We could poll for the API, but we've already written the logic in ansible to continue retrying the restart of the controller a few times and it works well enough.

Reply:
That's not going to work in static pods, which is our goal for 3.10 anyway.  Being unable to restart the static pod is going to be much less likely (since we're just terminating it and relying on the kubelet to bring it back), so I don't know that we should even have a retry loop at this level for the new path.